### PR TITLE
[Wallet] Add TransactionReview screen for CELO rewards

### LINKED ISSUE #6136: Valora activity page accurately represents cusd rewards to user

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Adds a "content" on the `TransactionReview` screen for the CELO rewards transfers. If the transaction comes from an address we know is used for CELO rewards, we show this content instead of the standard transaction one.

Screen Shot 2021-01-06 at 17 30 25

The "Learn More" button opens a `WebView` with a yet not implemented webpage.

### Other changes

- The `UserSection` can be non-expandable now and doesn't render the phone number text if there is no phone number (it was rendered empty before).

### Tested

Yes

### Related issues

- Fixes #6136

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/mobile/src/identity/saga.ts [hunk 1]
```diff
@@ -129,6 +130,10 @@ function* fetchKnownAddresses() {
   }
 }
 
+export function isCeloRewardSender(address: string) {
+  return store.getState().identity.addressToDisplayName[address]?.isCeloRewardSender
+}
+
```
Comment (jeanregisser) [inline packages/mobile/src/identity/saga.ts]: This bypasses usual react <-> redux communication and can lead to component not being updated on state change.
We should replace this with a selector.

File: packages/mobile/__mocks__/src/redux/store.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const store = {
+  getState: jest.fn(),
+}
```
Comment (jeanregisser) [inline packages/mobile/__mocks__/src/redux/store.ts]: This shouldn't be needed once `isCeloRewardSender` is changed to a selector.

**Review by jeanregisser — CHANGES_REQUESTED** [review]
See main concern below. I understand the short timing here, happy to chat about our options.

**Comment by calrobertlee** [comment]
Updated designs with slight copy changes
image

**Comment by gnardini** [comment]
Updated to follow the changes to the UI and the copy changes

File: packages/mobile/src/transactions/actions.ts [hunk 3]
```diff
@@ -138,7 +140,9 @@ export const navigateToPaymentTransferReview = (
       headerText = i18n.t('walletFlow5:transactionHeaderEscrowSent')
       break
     case TokenTransactionType.Received:
-      headerText = i18n.t('walletFlow5:transactionHeaderReceived')
+      headerText = addressToDisplayName[confirmationProps.address || '']?.isCeloRewardSender
+        ? i18n.t('walletFlow5:transactionHeaderCeloReward')
+        : i18n.t('walletFlow5:transactionHeaderReceived')
```
Comment (jeanregisser) [inline packages/mobile/src/transactions/actions.ts:145]: Nothing to do now, but I now realise this header text logic being in the action is what is causing these hoops with passing more of the redux state. It should probably be moved out of this and dealt in the screen/component displaying this.

**Review by jeanregisser — APPROVED** [review]
:shipit: 👍

File: packages/mobile/src/transactions/actions.ts (hunk 3 above)
Comment (gnardini) [reply packages/mobile/src/transactions/actions.ts:145]: Yeah, it does look a bit strange having it like this. I opened an issue to track this here: #6406

**Comment by Lss-Ankit** [comment]
Hi @gnardini We are block to get the rewards &  not able to buy the cUSD because user should redirect to add funds page by tapping on buy cUSD button on simplex page 
However. we are using mainnet for valora here user redirecting to simplex page for buying cUSD So, we are blocked to buy the cUSD 
Can you please do the needful that how we can buy the cUSD.
Thanks!

**Comment by Lss-Ankit** [comment]
Hi @gnardini We are block to get the rewards & not able to buy the cUSD because user should redirect to add funds page by tapping on Add cUSD button on celo.org.
Can you please help us how to Add cUSD.
Thanks!

**Pull request merged**